### PR TITLE
Fast datapath documentation

### PR DESCRIPTION
**Pull request by awh**
This seemed the best vehicle for receiving comments! @dpw @rade @inercia

### CODE REVIEW

File: docs/fdp-overview.markdown [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# Overlay Method Selection
```
Comment (inercia) [inline docs/fdp-overview.markdown]: Maybe you could include a short introduction with a description of the two overlays before jumping to the explanation of how we select one of them...

File: docs/fdp-overview.markdown [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+acknowledgement will cause the peers to fall back to the Sleeve
+overlay, which has a more sophisticated dynamic mechanism for coping
+with low path MTUs.
+
+To avoid triggering this fallback in typical deployments, the datapath
+interface is statically configured with an MTU of 1410 bytes allowing
+it to work with most underlay network provider MTUs, including GCE at
+1460 bytes (the fifty byte difference accomodates the vxlan overhead).
+
+# Virtual Ports
```
Comment (inercia) [inline docs/fdp-overview.markdown]: And what about moving this section to the introduction? It is some kind of glossary...

File: docs/fdp-overview.markdown [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+* Flows referring to netdev vports which have been removed
+* Flows created by forwarders which have been stopped
+* Flows related to MAC addresses which have not communicated recently
+
+Flows keyed on tunnel IPv4 address do not need to be cleared when a
+peer appears to change IP address due to NAT; this will cause a miss
+resulting in a new flow, and the old flow will expire naturally via
+the timer mechanism.
+
```
Comment (inercia) [inline docs/fdp-overview.markdown]: As a minor comment, maybe I would reorganize the sections. I would start with an _Introduction_ with the _Virtual Ports_. Then I would go into the FDP with the _Short Peer IDs_, _Overlay Method Selection_, _Misses and Flow Creation_ and _Flow Invalidation_. And I would finish with the advanced stuff: _PMTU Discovery_, _Local Bridging_ and _Heartbeats via vxlan_

And I really miss some [ASCII stuff](http://asciiflow.com/)... :wink:

File: docs/fdp-overview.markdown (hunk 1 above)
Comment (awh) [inline docs/fdp-overview.markdown]: Absolutely. There's a tentative plan to rework the existing 'how it works' document into an overarching piece that would include such a description, and then have detailed sections for fastdp and sleeve. That will probably happen in a separate PR though.

**Comment by awh** [comment]
Needs a section MAC expiry.

**Comment by awh** [comment]
Closing to retarget against master.

**Comment by dpw** [comment]
What happened to the part of this that didn't go into #1526?

**Pull request closed without merging**